packages/near-membrane-base/src/environment.ts

```diff
@@ -1,192 +1,206 @@
-                }
+            const configurable =
+                'configurable' in blueDescriptor ? !!blueDescriptor.configurable : undefinedSymbol;
+            const enumerable =
+                'enumerable' in blueDescriptor ? !!blueDescriptor.enumerable : undefinedSymbol;
+            const writable =
+                'writable' in blueDescriptor ? !!blueDescriptor.writable : undefinedSymbol;
+            const valuePointer =
+                'value' in blueDescriptor
+                    ? this.blueGetTransferableValue(blueDescriptor.value)
+                    : undefinedSymbol;
+            const getPointer =
+                'get' in blueDescriptor
+                    ? this.blueGetTransferableValue(blueDescriptor.get)
+                    : undefinedSymbol;
+            const setPointer =
+                'set' in blueDescriptor
+                    ? this.blueGetTransferableValue(blueDescriptor.set)
+                    : undefinedSymbol;
+            // installing descriptor into the red side
+            this.redCallableDefineProperty(
+                oPointer,
+                key,
+                configurable,
+                enumerable,
+                writable,
+                valuePointer,
+                getPointer,
+                setPointer
+            );
+        }
+    }
 
-                if (typeof blueDescriptor.set === 'function') {
-                    redDescriptor.set = function set(v: RedValue): void {
-                        // if a global setter is invoke, the value will be use as it
-                        // is as the result of the getter operation
-                        currentBlueGetter = () => v;
-                    };
+    lazyRemap(o: ProxyTarget, keys: PropertyKey[]) {
+        const enumerablePropertyKeys = ObjectKeys(o); // except symbols
+        const oPointer = this.blueGetTransferableValue(o) as Pointer;
+        const args: Parameters<CallableInstallLazyDescriptors> = [oPointer];
+        for (let i = 0, len = keys.length; i < len; i += 1) {
+            const key = keys[i];
+            let isEnumerable = true;
+            if (typeof key === 'symbol') {
+                if (ObjectPropertyIsEnumerable.call(o, key)) {
+                    isEnumerable = true;
                 }
             }
-            // "as any" supresses the "Type 'symbol' cannot be used as an index type."
-            redDescriptors[key as any] = redDescriptor;
+            // Skip index keys for magical descriptors of frames on the window proxy.
+            // TODO: this applies to all objects rather than just `window`, it might
+            // be a problem.
+            else if (RegExpTest(frameGlobalNamesRegExp, key as string)) {
+                // eslint-disable-next-line no-continue
+                continue;
+            } else {
+                isEnumerable = ArrayIncludes.call(enumerablePropertyKeys, key);
+            }
+            ArrayPush.call(args, key, isEnumerable);
```

Comment:
`ArrayPush.call` should not be using the `.call` of the prototype. We should use `ReflectApply`.